addons/payment/models/res_company.py

```diff
@@ -22,13 +22,60 @@ class ResCompany(models.Model):
     @api.model
     def action_open_payment_onboarding_payment_acquirer(self):
         """ Called by onboarding panel above the customer invoice list. """
-        # Fail if there are no existing accounts
+        menu_id = self.env.ref('payment.payment_acquirer_menu').id
+        return self._run_payment_onboarding_step('payment', menu_id)
+
+    @api.model
+    def _run_payment_onboarding_step(self, onboarding_module, menu_id):
+        """ Install the suggested payment modules, configure the acquirers, and mark the step done.
+
+        :param str onboarding_module: The module for which the payment onboarding step is run
+        :param int menu_id: The menu from which the user started the onboarding step, as an
+                            `ir.ui.menu` id
+        :return: The action returned by `action_stripe_connect_account`
+        :rtype: dict
+        """
         self.env.company.get_chart_of_accounts_or_fail()
```

Comment:
Since everything else is being rewritten, it might be a good idea to explicitely note that we're just checking that the current (user's?) company has a CoA, and to document that as part of `_run_payment_onboarding_step`.